src/utils/debounce.ts

```diff
@@ -0,0 +1,29 @@
+
+export function debounce(fn: Function, wait?: number, leading?: boolean) {
+  let timer;
+  let lastCalledAt = -Infinity;
+
+  function handleLeading(thisArg, args) {
```

Comment:
```ts
function handleLeading(thisArg: unknown, args: Parameters<T>)
```

Maybe it's better to move this function into the returned `debouncedFn` block?

Reply:
agreed, prevent duplication